src/server/system_services/account_server.js

```diff
@@ -475,14 +475,17 @@ function update_account(req) {
     if (params.ips && !_.every(params.ips, ip_range => (net.isIP(ip_range.start) && net.isIP(ip_range.end)))) {
         throw new RpcError('FORBIDDEN', 'Non valid IPs');
     }
-
+    if (req.rpc_params.nsfs_account_config && _.isUndefined(req.rpc_params.nsfs_account_config.uid) &&
+            _.isUndefined(req.rpc_params.nsfs_account_config.gid) && !req.rpc_params.nsfs_account_config.new_buckets_path) {
+                throw new RpcError('FORBIDDEN', 'Invalid nsfs_account_config');
+    }
     let updates = {
         name: params.name,
         email: params.new_email,
         next_password_change: params.must_change_password === true ? new Date() : undefined,
         allowed_ips: (!_.isUndefined(params.ips) && params.ips !== null) ? params.ips : undefined,
         preferences: _.isUndefined(params.preferences) ? undefined : params.preferences,
-        nsfs_account_config: req.rpc_params.nsfs_account_config
+        nsfs_account_config: req.rpc_params.nsfs_account_config && _.assign(account.nsfs_account_config, req.rpc_params.nsfs_account_config)
```

Comment:
@romayalon You shouldn't be trying to update the actual account which is in system store.
Instead, create a new object like this -
```
        nsfs_account_config: req.rpc_params.nsfs_account_config && {
            // first take existing configs from account
            ...account.nsfs_account_config,
            // then override with request configs
            ...req.rpc_params.nsfs_account_config
        }
```

Reply:
the reason why updating the account directly is bad is that this update was not yet applied to the system store, and we don't want any other request during that time to see that change.
I think at some point we will want to make the system store objects frozen (`js_utils.deep_freeze`) which will cause such calls to fail (on runtime).

Reply:
thanks for your input, I fixed it in https://github.com/noobaa/noobaa-core/pull/6542